### ISSUE TITLE
Saving and clearing filters

### ISSUE THREAD

**Issue by tkeuneman** [TerriaJS/magda#373]
- [x] As a user, if I make one change to a filter and I hit clear, it clears everything for that filter
- [x] As a user, if I apply a change to a filter and then click away from the module, the change I made is applied to that filter
- [x] If I hit tab, it applies the changes (more research is needed)
- [x] As a user, if I make a second input to a filter and I hit clear, it clears everything from that filter

<img width="585" alt="magda_-_search_-_v1_1_preview_mode_-_invision" src="https://user-images.githubusercontent.com/2201814/34854852-618df9be-f790-11e7-92e3-9f77ae80f3b9.png">

<img width="585" alt="magda_-_search_-_v1_1_preview_mode_-_invision" src="https://user-images.githubusercontent.com/2201814/34854934-08844be2-f791-11e7-8bca-865325e0a9af.png">

**Comment by rehaab** [comment]
@tkeuneman @philipgrimmett @chloeleichen 

I did some research on this. There isn't much per se on the accessibility of dropdown filter options - closest option was to look at form input styles, in which it is obligatory to have a "Submit" button (in our case, the "Apply" button) - unless the results dynamically update as options are selected. (Don’t think that happens in our case though)
It is especially important to have that for keyboard navigation, where users can click ‘Enter’ on ‘Apply’ to apply selected options, then hit escape to exit the dropdown or tab to move onto the next filter criteria.

In addition, currently the accessibility guidelines ask to avoid using color alone to indicate that an option is a button, or that an option has been selected (which is why checkboxes are encouraged). In our case, the Apply option may need an underline or an outline to indicate that it is a button - since currently it relies on colour and positioning alone. 

The click-outside-the-box design pattern is usually used to close modal boxes, and may be important in our case since we have no close icon on the actual filter. It is much more costly to lose all of ones filters by clicking outside the box (without having pressed apply) than to have accidentally applied filters which can be removed in one click. In my opinion, best practise now would be to save filters when users click outside the box (or hit escape) and display a message saying that filters have been applied - which will also help with accessibility. Ideally we should carry out a round of user testing on this. 

TL;DR: Should probably keep both for now then test.

**Comment by chloeleichen** [comment]
sounds good, thanks @rehaab after thinking about it overnight, i have come to agree with @tkeuneman and @rehaab on click somewhere else to apply selected filter, it makes the best sense.

**Comment by tkeuneman** [comment]
Sounds good Chloe and Rehaab!!
